presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ExtractSpatialJoins.java

```diff
@@ -148,24 +154,27 @@
 public class ExtractSpatialJoins
 {
     private static final TypeSignature GEOMETRY_TYPE_SIGNATURE = parseTypeSignature("Geometry");
+    private static final TypeSignature SPHERICAL_GEOGRAPHY_TYPE_SIGNATURE = parseTypeSignature("SphericalGeography");
     private static final String KDB_TREE_TYPENAME = "KdbTree";
 
     private final Metadata metadata;
     private final SplitManager splitManager;
     private final PageSourceManager pageSourceManager;
+    private final SqlParser sqlParser;
```

Comment:
I don't like this new dependency (SqlParser), which is being used just for the purpose of passing providing it to the method that will compute the type of an expression. We should consider adding an an abstraction to do that (which will contain the sql parser and anything else necessary inside).